src/systems/userinput/devices/vive-controller.js

```diff
@@ -59,16 +72,22 @@ export class ViveControllerDevice {
       }
     }
     const el = document.querySelector(this.selector);
-    if (el.components["tracked-controls"].controller !== this.gamepad) {
-      el.components["tracked-controls"].controller = this.gamepad;
-      el.setAttribute("tracked-controls", "controller", this.gamepad.index);
-    }
     const rayObject = el.object3D;
     rayObject.updateMatrixWorld();
     this.rayObjectRotation.setFromRotationMatrix(rayObject.matrixWorld);
     this.pose.position.setFromMatrixPosition(rayObject.matrixWorld);
     this.pose.direction.set(0, 0, -1).applyQuaternion(this.rayObjectRotation);
     this.pose.fromOriginAndDirection(this.pose.position, this.pose.direction);
     frame[this.path.pose] = this.pose;
+
+    if (this.gamepad.pose.position && this.gamepad.pose.orientation) {
```

Comment:
Even if we don't have both position and rotation we should still probably update what we do have.